ibis/common/bases.py

```diff
@@ -141,41 +139,38 @@ class Comparable(Abstract):
 
     Since the class holds a global cache of comparison results, it is important
     to make sure that the instances are not kept alive longer than necessary.
-    This is done automatically by using weak references for the compared objects.
     """
 
-    __cache__ = WeakCache()
-
-    def __eq__(self, other) -> bool:
-        try:
-            return self.__cached_equals__(other)
-        except TypeError:
-            return NotImplemented
+    __cache__ = {}
```

Comment:
Why is okay to not use weak keys here anymore?

Reply:
Because we remove the cached node ids when the node gets `__del__`-eted.